test/specs/default-controls.test.js

```diff
@@ -0,0 +1,371 @@
+      expect(instance.getDotIndexes(6, 3, 5, 'right')).toEqual([0, 3, 5]);
+      expect(instance.getDotIndexes(6, 1, 3, 'right')).toEqual([
+        0,
+        1,
+        2,
+        3,
+        4,
+        5
+      ]);
+
+      // testing extreme scenarios on number of pages when cellAlign = `center`
+      expect(instance.getDotIndexes(11, 5, 3, 'center')).toEqual([0, 5, 10]);
+      expect(instance.getDotIndexes(11, 2, 7, 'center')).toEqual([
+        0,
+        2,
+        4,
+        6,
+        8,
+        10
+      ]);
+
+      // testing extreme scenarios on number of pages when cellAlign = `right`
+      expect(instance.getDotIndexes(11, 5, 3, 'right')).toEqual([0, 5, 10]);
+      expect(instance.getDotIndexes(11, 2, 7, 'right')).toEqual([
+        0,
+        2,
+        4,
+        6,
+        8,
+        10
+      ]);
+
+      // testing edge case scenarios when cellAlign = `center`
+      expect(instance.getDotIndexes(5, 2, 6, 'center')).toEqual([0, 2, 4]);
+      expect(instance.getDotIndexes(5, 6, 2, 'center')).toEqual([0, 4]);
+      expect(instance.getDotIndexes(5, 5, 5, 'center')).toEqual([0, 4]);
+
+      // testing edge case scenarios when cellAlign = `right`
+      expect(instance.getDotIndexes(5, 2, 6, 'right')).toEqual([0, 2, 4]);
+      expect(instance.getDotIndexes(5, 6, 2, 'right')).toEqual([0, 4]);
+      expect(instance.getDotIndexes(5, 5, 5, 'right')).toEqual([0, 4]);
+
+      // testing if `center` and `right` cellAlign is disabled when slidesToScroll === 1
+      expect(instance.getDotIndexes(6, 2, 1, 'center')).toEqual([0, 2, 4, 5]);
+      expect(instance.getDotIndexes(6, 2, 1, 'right')).toEqual([0, 2, 4, 5]);
+    });
+  });
+});
+
+describe('<NextButton />', () => {
+  describe('#nextButtonDisable for cellAlign = `left`', () => {
+    let instance;
+
+    beforeEach(async () => {
+      const wrapper = mount(<NextButton buttonDisable={() => null} />);
+      instance = wrapper.instance();
+    });
+
+    // testing smaller number of pages
+    it('should return true for slide index >= 3', () => {
+      const alignLeftIndex2 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 2,
+        cellAlign: 'left',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(alignLeftIndex2)).toEqual(false);
+
+      // past the slide 3 should return true and disable the button
+      const testLeftIndex3 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 3,
+        cellAlign: 'left',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(testLeftIndex3)).toEqual(true);
+
+      const testLeftIndex4 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 4,
+        cellAlign: 'left',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(testLeftIndex4)).toEqual(true);
+    });
+
+    // testing extreme scenarios
+    it('should return true for slide index >= 4', () => {
```

Comment:
I would add a little more explanation of this test case to the `it` statement, because we don't expect the return value to be true _always_ when `slideIndex >= 4`. So something like, `...when "the other conditions are...etc, etc`. I know this can get a bit wordy, so just see if you could come up with a way to differentiate this test for the edge case it's being used to test